connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceTask.java

```diff
@@ -294,5 +302,9 @@ boolean update(long upstreamOffset, long downstreamOffset) {
             previousDownstreamOffset = downstreamOffset;
             return shouldSyncOffsets;
         }
+
+        void reset() {
+            shouldSyncOffsets = false;
```

Comment:
I was wondering if we should also move the updates for the `lastSyncUpstreamOffset` and `lastSyncDownstreamOffset` fields to this method, but there doesn't appear to be any downsides to this approach given our current usages of `update` and `reset`. 👍